homeassistant/components/media_player/spotify.py

```diff
@@ -328,3 +368,44 @@ def supported_features(self):
     def media_content_type(self):
         """Return the media type."""
         return MEDIA_TYPE_MUSIC
+
+
+class SpotifyPlaylistView(HomeAssistantView):
+    """View to retrieve Spotify Playlists."""
+
+    url = '/api/spotify/playlists'
+    name = "api:spotify/playlists"
+
+    @callback
+    def get(self, request):
+        """Retrieve playlists."""
+        return self.json(request.app['hass'].data[DOMAIN].playlists)
+
+
+class PlaylistData:
+    """Class to hold Spotify playlist data."""
+
+    def __init__(self, hass):
+        """Initialize the Spotify playlist store list."""
+        self.hass = hass
+        self.playlists = {}
+
+    def update(self, playlist=None):
+        """Add a playlist."""
+        if playlist is not None:
+            self.playlists[playlist['id']] = playlist
+        self.save()
+        self.playlists = self.load()
```

Comment:
Why this need to reload a playlist that was _just_ saved ?

Reply:
It can be removed, but I just wanted to avoid any differences between json and class data. Which approach would you choose? Just load once at init?

Reply:
isn't save() enough ?

from your code there is no difference...

Reply:
But then I need to call load() on init, because the idea was to save it locally to avoid all playlists will be discovered at each restart of HASS.

Reply:
might be a semantic issue... are you updating the persistent file or the playlist variable ? maybe splitting update into two different methods ?

Reply:
Why not like that?
https://gist.github.com/tringler/a48e8932e7e2692ce1546364df4cded8

Reply:
need to indent lines 20:21 and 25:26, but that looks good